### PR TITLE
Book Store: Connect the app with the bookstore API

### PR DESCRIPTION
**Pull request by raihan2bd**
# Book Store 🚀 
### Hi there, 👋 In This PR: 
- [ ] At first, I created a new branch then I added modal components to show the modal with HTTP responses ✅ 
- [ ] Then I added new functionality and split my actions with multiple files. ✅ 
- [ ] Then I added fetch and remove book functionality using redux actions and reducer ✅ 

***Thank you so much for visiting my pull request 👍***

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,49 +1,37 @@
...
+      id: payload.item_id,
+      title: payload.title,
+      author: payload.author,
+      category: payload.category,
+    };
+    const updatedState = { ...state, books: [...state.books, { ...book }] };
+    return updatedState;
+  });
   builder.addDefaultCase((state) => state);
 });
```
Comment (franclobo) [inline src/redux/books/books.js]: - [ ] Please 🙏🏽 use the [ducks pattern for redux](https://github.com/erikras/ducks-modular-redux) it avoids adding {actionTypes, actions, reducer} tuples for each use case and keeping these in separate files and even separate folders.

File: src/redux/books/bookActions.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      thunkAPI.dispatch(uiActions.successModal(response.data));
+      return id;
+    } catch (err) {
+      thunkAPI.dispatch(
+        uiActions.rejectModal(err.message || 'Failed to delete the book!!'),
+      );
+    }
+    return null;
+  },
+);
```
Comment (franclobo) [inline src/redux/books/bookActions.js]: - [ ] Please 🙏🏽 use the [ducks pattern for redux](https://github.com/erikras/ducks-modular-redux) it avoids adding {actionTypes, actions, reducer} tuples for each use case and keeping these in separate files and even separate folders.
- [ ] I couldn't find the API URL in your file, please 🙏🏽 use it to fetch the data.

File: src/index.js [hunk 3]
```diff
@@ -2,10 +2,13 @@ import React from 'react';
 import ReactDOM from 'react-dom/client';
 import { BrowserRouter } from 'react-router-dom';
 import { Provider } from 'react-redux';
+import axios from 'axios';
 
 import store from './redux/configureStore';
 import App from './App';
 
+axios.defaults.baseURL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/eZNPE2dPLtWHMUhnsTBH/books';
```
Comment (franclobo) [inline src/index.js]: - [ ] Please 🙏🏽 refactor your code and use the API URL in the corresponding redux file to have a clean code.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @raihan2bd, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- No linter errors ✔️ 
- Correct Git flow ✔️ 
- Descriptive PR ✔️ 
- Professional README file ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Remember we spend more time reading code than writing, please 🙏🏽 use the duck pattern for redux it is cleaner and easier to understand.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 3 above)
Comment (raihan2bd) [reply src/index.js]: Hi, @franclobo 👋  Thanks a lot for your valuable suggestion I will apply your suggestions before the next review 👍

**Review by DammyShittu — APPROVED** [review]
Hi @raihan2bd,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**